react/Select/index.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+import ReactSelect from 'react-select'
+import styles from './styles.styl'
+import Icon from '../Icon'
+import palette from '../../stylus/settings/palette.json'
+
+const withCustomControl = CustomContol => ({ innerProps, children }) => (
+  <div {...innerProps}>
+    {CustomContol}
+    <div className={styles['select-control__input']}>{children}</div>
+  </div>
+);
+
+const Option = ({ label, isSelected, innerProps, withCheckbox }) => (
+  <div {...innerProps} className={classNames(styles['select-option'], { [styles['select-option--selected']]: isSelected && !withCheckbox })}>
+    { withCheckbox &&
+        <input
+        type="checkbox"
+        checked={isSelected}
+        className={styles['select-option__checkbox']}
+      />
+    }
+    {label}
+    { (isSelected && !withCheckbox) && <Icon icon="check-circleless" color={palette['dodgerBlue']} className={styles['select-option--checkmark']} /> }
+  </div>
+);
+
+Option.propTypes = {
+  label: PropTypes.string.isRequired,
+  isSelected: PropTypes.bool.isRequired,
+  innerProps: PropTypes.object.isRequired,
+  withCheckbox: PropTypes.bool,
+}
+
+Option.defaultProps = {
+  withCheckbox: false
+}
+
+const CheckboxOption = ({ ...props }) => (
+  <Option {...props} withCheckbox />
+);
+
+CheckboxOption.propTypes = {}
```

Comment:
Why not copy Option proptypes ?

Reply:
Weeeell. I'm not even sure about the `Option proptypes`. Most of the props defined here (such as `innerProps`, `label`, ect) are actually injected by `react-select` and not by the user. The only addition is the `withCheckbox`. You're also never really supposed to instanciate these components, you just pass their definition to `react-select`.

So, should we describe the props that `react-select` uses? I'd say probably not, and then `CheckboxOption` has no props, and `Option` has only the `withCheckbox` prop. What do you think?

Reply:
Ah yes OK, sorry I was confused. I agree.